### PR TITLE
Balances

### PR DESCRIPTION
**Pull request by dvd-schwrtz**

### CODE REVIEW

File: src/issuance.ts [hunk 1]
```diff
@@ -285,33 +286,55 @@ export function handleTransferSNX(event: SNXTransferEvent): void {
...
+  value: BigInt,
+  isOutbound: boolean,
+): void {
+  let totalBalance = ZERO;
+  let aggregateSynthBalanceID = account.toHex() + '-' + synth;
+  let aggregateSynthBalance = AggregateSynthBalance.load(aggregateSynthBalanceID);
+
+  if (aggregateSynthBalance == null) {
+    let newAggregateSynthBalance = new AggregateSynthBalance(aggregateSynthBalanceID);
+    newAggregateSynthBalance.amount = contract.balanceOf(account);
```
Comment (dvd-schwrtz) [inline src/issuance.ts:303]: note this should make it work for Optimism even if they do a regenesis

File: src/issuance.ts [hunk 2]
```diff
@@ -285,33 +286,55 @@ export function handleTransferSNX(event: SNXTransferEvent): void {
...
     let currencyKeyTry = contract.try_currencyKey();
     if (!currencyKeyTry.reverted) {
-      source = currencyKeyTry.value.toString();
+      synth = currencyKeyTry.value.toString();
     }
   }
 
-  if (event.params.from.toHex() != ZERO_ADDRESS.toHex()) {
-    trackSynthHolder(contract, source, event.params.from);
+  if (event.params.from.toHex() != ZERO_ADDRESS.toHex() && event.params.from.toHex() != FEE_ADDRESS.toHex()) {
```
Comment (dvd-schwrtz) [inline src/issuance.ts:333]: don't track the fee address or the null address as a user with balances

File: subgraphs/issuance.graphql [hunk 3]
```diff
@@ -68,15 +68,20 @@ type DebtSnapshot @entity {
...
-  " currencyKey of the synth "
+  amount: BigInt!
+  account: Bytes!
+  timestamp: BigInt!
   synth: String!
+}
 
-  " units of the synth held in wei "
-  balanceOf: BigDecimal!
+" we dont query this entity but only use it to store aggregate data we need during syncing "
```
Comment (dvd-schwrtz) [inline subgraphs/issuance.graphql:80]: decided to go this route after having a few issues with the derivedFrom method which worked great on shorts but not as much here.

**Comment by dvd-schwrtz** [comment]
Should be ready for review, but let's merge tomorrow. I am syncing the latest changes and just want to make 100% sure it is working properly before we merge. I tested a similar version in the data library and all appears to be working ok

**Review by noahlitvin — APPROVED** [review]

**Pull request merged**